app/com/lucidchart/open/cashy/amazons3/S3Client.scala

```diff
@@ -158,4 +160,15 @@ class S3Client {
     }
   }
 
-}
\ No newline at end of file
+}
+
+trait AWSConfig {
+  protected def getAWSCredentials(): AWSCredentials = {
+    try {
+      (new InstanceProfileCredentialsProvider()).getCredentials()
+    } catch {
+      case e: Exception =>
+          (new ProfileCredentialsProvider("/etc/aws/dev-credentials", "default")).getCredentials()
+    }
+  }
```

Comment:
The problem with your solution, @pauldraper, is when the `/etc/aws/dev-credentials` file doesn't exist (like in production). For workstations, we want to use that file, but in production, we want to use the instance profile. The ProfileCredentialsProvider throws an exception if that file doesn't exist, or is empty. It's a pretty crappy class.